modxcms/revolution
# Make date display consistently across manager components

### What does it do?

1. Adds a new Formatter class and updated several processor classes (and a couple controllers) to use the new date formatting methods provided. 
2. Adds a timestamp validation static method on the main modX class.
3. Includes missing language topic for the Resource data page (currently two column headings are missing because the manager_log topic needs to be included).

### Why is it needed?

The manager UI is inconsistent in its formatting of dates and times. This PR consolidates the formatting logic in one place and applies it to every component that displays a date (unless I've overlooked something). 

### How to test

1. As this PR includes two new system settings, a data rebuild is necessary (`/_build/transport.core.php`) then a reinstall (upgrade) via `/setup`.
2. Set a `manager_date_format` and `manager_time_format`, as well as a `manager_datetime_separator` in the system settings.
3. Take a run through the entire interface and verify that areas that display a date/time do so as expected and match your expected format everywhere. Note that there are two places where these changes are purposely no applied: The system info page (where the server and local times are shown) and the manager error log (where it's impractical to try to alter the entry format).

### Related issue(s)/PR(s)
Resolves #14961. 
Resolves #16512.
